addons/base_vat/models/res_partner.py

```diff
@@ -168,27 +175,30 @@ def check_vat(self):
                 msg = partner._build_vat_error_message(country and country.code.lower() or None, partner.vat, partner_label)
                 raise ValidationError(msg)
 
-    @api.onchange('vat', 'country_id')
-    def _onchange_check_vies(self):
-        """ Check the VAT number with VIES, if enabled. Return a non-blocking warning if the check fails."""
-        if self.env.context.get('company_id'):
-            company = self.env['res.company'].browse(self.env.context['company_id'])
-        else:
-            company = self.env.company
-        if not company.vat_check_vies:
+    @api.depends('vies_vat_to_check')
+    def _compute_vies_valid(self):
+        """ Check the VAT number with VIES, if enabled."""
+        if not self.env['ir.config_parameter'].get_param('base_vat.perform_vies_check'):
+            self.write({'vies_valid': False})
```

Comment:
Do not call `write` explicitly, it might have unwanted side effects.
`__set__` also works on recordsets of size >= 2
```suggestion
            self.vies_valid = False
```